### PR TITLE
now using #pragma instead of header guards, fixed spacing formating, too, on all H's @Hydroque

### PR DESCRIPTION
**Pull request by Hydroque**
Check if compiles successfully before using, just in case I overwritten something that depended on a `#define` Just simple spacing fixes so things aren't crowded together with everything .h converted to `#pragma`. I didn't touch anything from `freeablo/extern` see #325

### CODE REVIEW

**Comment by Hydroque** [comment]
Travis failed to launch the clang test, which error not during build, but during setup to build.

**Comment by Predelnik** [comment]
It actually failed because now there is enforced `clang-format` on project since a little while ago. You should just run this line locally in your bash:

     curl http://termbin.com/9hje > /tmp/fa_format_diff.patch && git apply /tmp/fa_format_diff.patch

And attach changes to your pull request.

File: extern/nativefiledialog/src/simple_exec.h [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // copied from: https://github.com/wheybags/simple_exec/blob/5a74c507c4ce1b2bb166177ead4cca7cfa23cb35/simple_exec.h
+// modified #ifndef SIMPLE_EXEC_H header guard to #pragma once
```
Comment (wheybags) [inline extern/nativefiledialog/src/simple_exec.h]: This is a mistake?

Comment (Hydroque) [reply extern/nativefiledialog/src/simple_exec.h]: Yes. This slipped through during my decision to not touch extern.

**Comment by Hydroque** [comment]
Since how long ago? I immediately pulled, fixed, and pushed to the branch without having to ff. I'm not sure how to amend a pull request yet, today is my Christmas party, I'll get this squared away tomorrow.

**Comment by Hydroque** [comment]
```
C:\git\freeablo>git apply tmp\fa_format_diff.patch
error: patch failed: apps/freeablo/engine/engineinputmanager.h:64
error: apps/freeablo/engine/engineinputmanager.h: patch does not apply
error: patch failed: apps/freeablo/engine/enginemain.h:55
error: apps/freeablo/engine/enginemain.h: patch does not apply
error: patch failed: apps/freeablo/engine/inputobserverinterface.h:45
error: apps/freeablo/engine/inputobserverinterface.h: patch does not apply
error: patch failed: apps/freeablo/engine/threadmanager.h:58
error: apps/freeablo/engine/threadmanager.h: patch does not apply
error: patch failed: apps/freeablo/faaudio/audiomanager.h:44
error: apps/freeablo/faaudio/audiomanager.h: patch does not apply
error: patch failed: apps/freeablo/fagui/autocomplete.h:51
error: apps/freeablo/fagui/autocomplete.h: patch does not apply
error: patch failed: apps/freeablo/fagui/commandhistory.h:27
error: apps/freeablo/fagui/commandhistory.h: patch does not apply
error: patch failed: apps/freeablo/fagui/dialogmanager.h:71
error: apps/freeablo/fagui/dialogmanager.h: patch does not apply
error: patch failed: apps/freeablo/fagui/guimanager.h:140
error: apps/freeablo/fagui/guimanager.h: patch does not apply
error: patch failed: apps/freeablo/fagui/menu/menuscreen.h:55
error: apps/freeablo/fagui/menu/menuscreen.h: patch does not apply
error: patch failed: apps/freeablo/fagui/menu/pausemenuscreen.h:35
error: apps/freeablo/fagui/menu/pausemenuscreen.h: patch does not apply
error: patch failed: apps/freeablo/fagui/menu/selectheromenuscreen.h:65
error: apps/freeablo/fagui/menu/selectheromenuscreen.h: patch does not apply
error: patch failed: apps/freeablo/fagui/menu/startingmenuscreen.h:26
error: apps/freeablo/fagui/menu/startingmenuscreen.h: patch does not apply
error: patch failed: apps/freeablo/fagui/menuhandler.h:53
error: apps/freeablo/fagui/menuhandler.h: patch does not apply
error: patch failed: apps/freeablo/fagui/nkhelpers.h:35
error: apps/freeablo/fagui/nkhelpers.h: patch does not apply
error: patch failed: apps/freeablo/fagui/textcolor.h:11
error: apps/freeablo/fagui/textcolor.h: patch does not apply
error: patch failed: apps/freeablo/falevelgen/levelgen.h:13
error: apps/freeablo/falevelgen/levelgen.h: patch does not apply
error: patch failed: apps/freeablo/falevelgen/mst.h:9
error: apps/freeablo/falevelgen/mst.h: patch does not apply
error: patch failed: apps/freeablo/falevelgen/random.h:31
error: apps/freeablo/falevelgen/random.h: patch does not apply
error: patch failed: apps/freeablo/falevelgen/tileset.h:187
error: apps/freeablo/falevelgen/tileset.h: patch does not apply
error: patch failed: apps/freeablo/farender/animationplayer.h:53
error: apps/freeablo/farender/animationplayer.h: patch does not apply
error: patch failed: apps/freeablo/farender/fontinfo.h:53
error: apps/freeablo/farender/fontinfo.h: patch does not apply
error: patch failed: apps/freeablo/farender/renderer.h:149
error: apps/freeablo/farender/renderer.h: patch does not apply
error: patch failed: apps/freeablo/farender/spritecache.h:145
error: apps/freeablo/farender/spritecache.h: patch does not apply
error: patch failed: apps/freeablo/farender/spritemanager.h:78
error: apps/freeablo/farender/spritemanager.h: patch does not apply
error: patch failed: apps/freeablo/fasavegame/gameloader.h:25
error: apps/freeablo/fasavegame/gameloader.h: patch does not apply
error: patch failed: apps/freeablo/fasavegame/objectidmapper.h:20
error: apps/freeablo/fasavegame/objectidmapper.h: patch does not apply
error: patch failed: apps/freeablo/faworld/actor.h:105
error: apps/freeablo/faworld/actor.h: patch does not apply
error: patch failed: apps/freeablo/faworld/actor/attackstate.h:23
error: apps/freeablo/faworld/actor/attackstate.h: patch does not apply
error: patch failed: apps/freeablo/faworld/actor/basestate.h:22
error: apps/freeablo/faworld/actor/basestate.h: patch does not apply
error: patch failed: apps/freeablo/faworld/actoranimationmanager.h:67
error: apps/freeablo/faworld/actoranimationmanager.h: patch does not apply
error: patch failed: apps/freeablo/faworld/actorstats.h:32
error: apps/freeablo/faworld/actorstats.h: patch does not apply
error: patch failed: apps/freeablo/faworld/behaviour.h:68
error: apps/freeablo/faworld/behaviour.h: patch does not apply
error: patch failed: apps/freeablo/faworld/faction.h:35
error: apps/freeablo/faworld/faction.h: patch does not apply
error: patch failed: apps/freeablo/faworld/findpath.h:21
error: apps/freeablo/faworld/findpath.h: patch does not apply
error: patch failed: apps/freeablo/faworld/gamelevel.h:100
error: apps/freeablo/faworld/gamelevel.h: patch does not apply
error: patch failed: apps/freeablo/faworld/hoverstate.h:34
error: apps/freeablo/faworld/hoverstate.h: patch does not apply
error: patch failed: apps/freeablo/faworld/inventory.h:132
error: apps/freeablo/faworld/inventory.h: patch does not apply
error: patch failed: apps/freeablo/faworld/item.h:298
error: apps/freeablo/faworld/item.h: patch does not apply
error: patch failed: apps/freeablo/faworld/itemmanager.h:60
error: apps/freeablo/faworld/itemmanager.h: patch does not apply
error: patch failed: apps/freeablo/faworld/itemmap.h:98
error: apps/freeablo/faworld/itemmap.h: patch does not apply
error: patch failed: apps/freeablo/faworld/movementhandler.h:44
error: apps/freeablo/faworld/movementhandler.h: patch does not apply
error: patch failed: apps/freeablo/faworld/player.h:45
error: apps/freeablo/faworld/player.h: patch does not apply
error: patch failed: apps/freeablo/faworld/playerfactory.h:26
error: apps/freeablo/faworld/playerfactory.h: patch does not apply
error: patch failed: apps/freeablo/faworld/position.h:46
error: apps/freeablo/faworld/position.h: patch does not apply
error: patch failed: apps/freeablo/faworld/world.h:129
error: apps/freeablo/faworld/world.h: patch does not apply
error: patch failed: components/audio/audio.h:24
error: components/audio/audio.h: patch does not apply
error: patch failed: components/cel/celdecoder.h:71
error: components/cel/celdecoder.h: patch does not apply
error: patch failed: components/cel/celfile.h:26
error: components/cel/celfile.h: patch does not apply
error: patch failed: components/cel/celframe.h:26
error: components/cel/celframe.h: patch does not apply
error: patch failed: components/cel/pal.h:36
error: components/cel/pal.h: patch does not apply
error: patch failed: components/diabloexe/affix.h:38
error: components/diabloexe/affix.h: patch does not apply
error: patch failed: components/diabloexe/baseitem.h:56
error: components/diabloexe/baseitem.h: patch does not apply
error: patch failed: components/diabloexe/characterstats.h:37
error: components/diabloexe/characterstats.h: patch does not apply
error: patch failed: components/diabloexe/diabloexe.h:91
error: components/diabloexe/diabloexe.h: patch does not apply
error: patch failed: components/diabloexe/monster.h:98
error: components/diabloexe/monster.h: patch does not apply
error: patch failed: components/diabloexe/npc.h:30
error: components/diabloexe/npc.h: patch does not apply
error: patch failed: components/diabloexe/uniqueitem.h:53
error: components/diabloexe/uniqueitem.h: patch does not apply
error: patch failed: components/faio/fafileobject.h:37
error: components/faio/fafileobject.h: patch does not apply
error: patch failed: components/faio/faio.h:63
error: components/faio/faio.h: patch does not apply
error: patch failed: components/input/common.h:30
error: components/input/common.h: patch does not apply
error: patch failed: components/input/hotkey.h:20
error: components/input/hotkey.h: patch does not apply
error: patch failed: components/input/inputmanager.h:91
error: components/input/inputmanager.h: patch does not apply
error: patch failed: components/input/keys.h:150
error: components/input/keys.h: patch does not apply
error: patch failed: components/level/baseitemmanager.h:11
error: components/level/baseitemmanager.h: patch does not apply
error: patch failed: components/level/baseproperty.h:22
error: components/level/baseproperty.h: patch does not apply
error: patch failed: components/level/dun.h:46
error: components/level/dun.h: patch does not apply
error: patch failed: components/level/level.h:104
error: components/level/level.h: patch does not apply
error: patch failed: components/level/min.h:20
error: components/level/min.h: patch does not apply
error: patch failed: components/level/sol.h:22
error: components/level/sol.h: patch does not apply
error: patch failed: components/level/tileset.h:22
error: components/level/tileset.h: patch does not apply
error: patch failed: components/misc/assert.h:23
error: components/misc/assert.h: patch does not apply
error: patch failed: components/misc/disablewarn.h:15
error: components/misc/disablewarn.h: patch does not apply
error: patch failed: components/misc/helper2d.h:22
error: components/misc/helper2d.h: patch does not apply
error: patch failed: components/misc/maxcurrentitem.h:23
error: components/misc/maxcurrentitem.h: patch does not apply
error: patch failed: components/misc/md5.h:94
error: components/misc/md5.h: patch does not apply
error: patch failed: components/misc/misc.h:108
error: components/misc/misc.h: patch does not apply
error: patch failed: components/misc/savePNG.h:32
error: components/misc/savePNG.h: patch does not apply
error: patch failed: components/misc/stdhashes.h:22
error: components/misc/stdhashes.h: patch does not apply
error: patch failed: components/nuklearmisc/inputfwd.h:13
error: components/nuklearmisc/inputfwd.h: patch does not apply
error: patch failed: components/nuklearmisc/standaloneguispritehandler.h:65
error: components/nuklearmisc/standaloneguispritehandler.h: patch does not apply
error: patch failed: components/nuklearmisc/widgets.h:8
error: components/nuklearmisc/widgets.h: patch does not apply
error: patch failed: components/render/levelobjects.h:44
error: components/render/levelobjects.h: patch does not apply
error: patch failed: components/render/misc.h:38
error: components/render/misc.h: patch does not apply
error: patch failed: components/render/render.h:138
error: components/render/render.h: patch does not apply
error: patch failed: components/render/rocketglue/drawcommand.h:34
error: components/render/rocketglue/drawcommand.h: patch does not apply
error: patch failed: components/render/sdl_gl_funcs.h:49
error: components/render/sdl_gl_funcs.h: patch does not apply
error: patch failed: components/serial/loader.h:70
error: components/serial/loader.h: patch does not apply
error: patch failed: components/serial/streaminterface.h:46
error: components/serial/streaminterface.h: patch does not apply
error: patch failed: components/serial/textstream.h:60
error: components/serial/textstream.h: patch does not apply
error: patch failed: components/settings/settings.h:128
error: components/settings/settings.h: patch does not apply
error: patch failed: components/statemachine/statemachine.h:73
error: components/statemachine/statemachine.h: patch does not apply
```

I get this as try to apply the patch with `git apply tmp\fa_format_diff.patch` whilst in the root directory of freeablo source. I guess the patch was useless. However, the master is now updated with the comment taken out. Request to merge.

File: extern/nativefiledialog/src/simple_exec.h (hunk 1 above)
Comment (Hydroque) [reply extern/nativefiledialog/src/simple_exec.h]: Done

**Comment by Lectem** [comment]
You probably had other changes in local, or a different commit from the one sent to Travis CI.
Try running `curl http://termbin.com/e2du > /tmp/fa_format_diff.patch && git apply /tmp/fa_format_diff.patch` on your https://github.com/wheybags/freeablo/pull/328/commits/e83e31d01edb8c80590fa2877fe93cc04e38ed48 commit

**Comment by Hydroque** [comment]
There was nothing fixed by changing anything. Time to scrap and try again.

**Comment by Lectem** [comment]
@Hydroque I applied the patch locally, and created #332 as you already deleted the branch.
It worked just fine, don't know why you couldn't apply it.
I simply checked out your commit and applied the patch as described.
In any case your changes should probably get into master now

**Comment by Hydroque** [comment]
The reason it wouldn't work is because I was using the wrong autopatch. I created a commit and I pushed the correct changes to it, but I had no way of getting the patch so I had to let it run to get it. Only the owner or something can restart a travis-cl build and its ready for merging on the new pr I made. The one that was just uploaded is a duplicate.

**Comment by Predelnik** [comment]
travis-ci should restart automatically on every change you push (possibly with -f) to PR branch, I never seen it fail at it, so that's a bit strange.

**Comment by Hydroque** [comment]
I wonder if its because I did it mid-build

**Pull request closed without merging**